### PR TITLE
Performed some code refactoring - code is building successfully.

### PR DESCRIPTION
**Pull request by Chanpreet-Singh**
Performed some code refactoring:
1. Pull-up variable
2. Push-down method
3. Extract class
4. Extract Method
5. Rename variable

### CODE REVIEW

**Review by akhunovaa — COMMENTED** [review]
Hello 👋 
x_coord, y_coord, z_coord are more readable?

**Comment by Chanpreet-Singh** [comment]
Hi,
Hope you're doing well.

Thank you so much for accepting my pull request and merging it into your
code.
It was a part of my university assignment.

Just because you accepted it, I got 2 marks. I am very thankful to you.

I was just doing the refactoring for the sake of an academic assignment -
had no idea that you will be so kind.
Regarding the rename part, I know it might be absurd but would request you
to please hold it till 25th April, after that if you wish to revert it,
please do it. I don't want to mess up with your hard work.

Again, thank you so much. Those 2 marks might help me pass the course in my
university and eventually save some money.

Would love to connect with you on LinkedIn:
https://www.linkedin.com/in/chanpreet-singh-ai/

Best wishes,
Chanpreet Singh

**Comment by akhunovaa** [comment]
no problem! If u wish, maybe later, it will be good to receive there new contributions from you ;)

File: src/main/java/ru/mycubecraft/engine/graph/CollisionPrediction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    public static void intersectSweptAabbAabb(int x_coord, int y_coord, int z_coord, float px, float py, float pz, float dx, float dy, float dz,
+                                        List<Contact> contacts, BlockField blockField) {
+        float pxmax = px + PLAYER_WIDTH, pxmin = px - PLAYER_WIDTH, pymax = py + PLAYER_HEIGHT - PLAYER_EYE_HEIGHT, pymin = py - PLAYER_EYE_HEIGHT,
+                pzmax = pz + PLAYER_WIDTH, pzmin = pz - PLAYER_WIDTH;
+
+        float xInvEntry = dx > 0f ? -pxmax : 1 - pxmin,
+                xInvExit = dx > 0f ? 1 - pxmin : -pxmax;
+
+        //boolean xNotValid = dx == 0 || blockField.load(x_coord + (dx > 0 ? -1 : 1), y_coord, z_coord) != null;
+        boolean xNotValid = dx == 0;
```
Comment (dandgerson) [inline src/main/java/ru/mycubecraft/engine/graph/CollisionPrediction.java:37]: There are naming convention in Java booleans with prefix `is/has`?

`boolean isXNotValid = dx == 0`

**Pull request merged**